src/lib.rs

```diff
@@ -165,6 +176,54 @@ impl CrashReporter {
             data.insert("arithmetic_mean".to_string(), json!(arithmetic_mean));
         }
 
+        if self.show_os_information {
+            let value = &mut data["errorGroups"];
+            let all_crashes: &mut Vec<serde_json::Value> = value.as_array_mut().unwrap();
+
+            // object: Object<Map<String, Value>>
+            for object in all_crashes.iter_mut() {
+                let crash = object.as_object_mut().unwrap();
+                let oses = crash["operating_systems"].as_array().unwrap();
+                let crash_amount_devices_overall = crash["deviceCount"].as_f64().unwrap();
+                let mut formatted = oses
+                    .iter()
+                    .filter(|os| {
+                        let crash_amount_os = os["errorCount"].as_f64().unwrap();
+                        let percentage =
+                            (crash_amount_os / crash_amount_devices_overall) as f64 * 100.0;
+                        percentage > 5.0
+                    })
+                    .map(|os| {
+                        let crash_amount_os = os["errorCount"].as_f64().unwrap();
+                        let percentage =
+                            (crash_amount_os / crash_amount_devices_overall) as f64 * 100.0;
+                        let os_string = os["operatingSystemName"].as_str().unwrap();
+                        format!(
+                            "{}: {} crashes ({:.2}%) ",
+                            os_string, crash_amount_os, percentage
+                        )
+                    })
+                    .enumerate()
+                    .fold(String::new(), |mut formatted, (idx, sub)| {
```

Comment:
Didn't know about fold, nice

Reply:
Not my idea ;)

I don't really like the whole chain. When I added debug information, it seems that the (whole) chain is executed item by item. I.e. there is no complete execution of filter, map, enumerate, fold separately. One item goes through the whole chain, then the next. Maybe this is only true for parts of the chain, but Rust tooling isn't really how I'm used to work. This behavior actually forced my to add the _cleanup_ in line [218](https://github.com/xing/recrep/blob/RELEASE-1952/show_operating_system_of_crash/src/lib.rs#L218), where is remove some previously added string and add the "and more" suffix.